ethereum/rpc/namespaces/eth/api.go

```diff
@@ -565,16 +567,11 @@ func (e *PublicAPI) doCall(
 		accessList = args.AccessList
 	}
 
-	// Set destination address for call
-	var fromAddr sdk.AccAddress
-	if args.From != nil {
-		fromAddr = sdk.AccAddress(args.From.Bytes())
-	} else {
-		fromAddr = sdk.AccAddress(common.Address{}.Bytes())
+	if args.From == nil {
 		args.From = &common.Address{}
 	}
 
-	_, seq, err := e.clientCtx.AccountRetriever.GetAccountNumberSequence(e.clientCtx, fromAddr)
+	seq, err := getAccountNonce(e.clientCtx, e.backend, *args.From, true)
```

Comment:
is it fine to consider pending tx there?

Reply:
no, you need to check the block number and see if it's the pending block value

Reply:
done

Reply:
done